src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/AnsiStringMarshaller.cs

```diff
@@ -44,8 +44,10 @@ public AnsiStringMarshaller(string? str, Span<byte> buffer)
                 return;
             }
 
+            int maxBytesNeeded = checked(Marshal.SystemMaxDBCSCharSize * str.Length);
+
             // >= for null terminator
-            if ((long)Marshal.SystemMaxDBCSCharSize * str.Length >= buffer.Length)
+            if (maxBytesNeeded >= buffer.Length)
```

Comment:
I have done this using `(long)` multiplication to avoid artificially reducing the maximum supported length for ASCII strings.

(I am ok with reducing the maximum supported length for ASCII strings. I am just pointing out that this change is turning working scenario into an exception.)

Reply:
My other reason for using the `long` multiplication was that it is cheaper than the one with overflow check.

Reply:
I see. I can put that back and will add a comment about why not checked.